src/sagas/dependency.saga.js

```diff
@@ -0,0 +1,325 @@
+// @flow
+
+import { select, call, put, cancel, takeEvery } from 'redux-saga/effects';
+import { getPathForProjectId } from '../reducers/paths.reducer';
+import {
+  loadProjectDependency,
+  loadProjectDependencies,
+} from '../services/read-from-disk.service';
+import {
+  installDependency,
+  uninstallDependency,
+  installDependencies,
+  uninstallDependencies,
+} from '../services/dependencies.service';
+import {
+  ADD_DEPENDENCY_START,
+  ADD_DEPENDENCY_ERROR,
+  ADD_DEPENDENCY_FINISH,
+  UPDATE_DEPENDENCY_START,
+  UPDATE_DEPENDENCY_ERROR,
+  UPDATE_DEPENDENCY_FINISH,
+  DELETE_DEPENDENCY_START,
+  DELETE_DEPENDENCY_ERROR,
+  DELETE_DEPENDENCY_FINISH,
+  ADD_DEPENDENCIES_START,
+  ADD_DEPENDENCIES_ERROR,
+  ADD_DEPENDENCIES_FINISH,
+  DELETE_DEPENDENCIES_START,
+  DELETE_DEPENDENCIES_ERROR,
+  DELETE_DEPENDENCIES_FINISH,
+  addDependencyFinish,
+  addDependencyError,
+  updateDependencyFinish,
+  updateDependencyError,
+  deleteDependencyFinish,
+  deleteDependencyError,
+  addDependenciesStart,
+  addDependenciesError,
+  addDependenciesFinish,
+  deleteDependenciesStart,
+  deleteDependenciesError,
+  deleteDependenciesFinish,
+  showNotification,
+  updateNotification,
+  completeNotification,
+  failNotification,
+} from '../actions';
+
+// Since redux-saga is intended to handle side-effects of actions,
+// all sagas are run *after* all reducers, even though they are
+// incorporated into redux as a middleware. As such, we cannot rely
+// on a store-based check for whether or not a package.json is locked,
+// as any locking actions will lock it in the store before our saga
+// has a chance to check if it is locked or not. As such, we must handle
+// monitoring the lock status within the saga, so we can control the
+// order of checking.
+//
+// NOTE: the previous reducer added false entries for new projects on
+// addition and for all projects after a refresh, but undefined is
+// falsey and has the same effect so I elected not to add an extra
+// saga for these actions.
```

Comment:
I think we can get rid of this comment; it's maybe better suited as a PR comment like this one? Since once this merges, folks won't know what the previous reducer was.